client/tracing/src/block/mod.rs

```diff
@@ -300,15 +298,15 @@ fn event_key_filter(event: &TraceEvent, storage_keys: &str) -> bool {
 		.unwrap_or(false)
 }
 
-/// Filter out spans that do not meet our targets and if the span is from WASM
-/// update its `name` and `target` fields to the WASM values for those fields.
+/// Filter out spans that do not match our targets and if the span is from WASM update its `name`
+/// and `target` fields to the WASM values for those fields.
 //
-// Due the original `tracing` crate requiring trace metadata to be static,
-// the workaround for substrate's WASM tracing wrappers is to put the `name`
-// and `target` in the `values` map (normally they would be in the static metadata.)
-// Here, if a special WASM name or target key is found in the `values` we remove
-// it and put the key value pair in the span's metadata, making it consistent
-// with spans that come from native.
+// The `tracing` crate requires trace metadata to be static. This does not work for wasm code in
+// substrate, as it is regularly updated with new code from on-chain events. The workaround for this
+// is for substrate's WASM tracing wrappers to put the `name` and `target` data in the `values` map
+// (normally they would be in the static metadata assembled at compile time). Here, if a special
+// WASM `name` or `target` key is found in the `values` we remove it and put the key value pair in
+// the span's metadata, making it consistent with spans that come from native code.
```

Comment:
I found this really helpful. @dvdplm Do you think it would make sense to add a modified version of this comment directly to the WASM tracing code as well.

Reply:
I think it bears repeating for sure. Do you know of a specific place to put it? Please do!

Reply:
See https://github.com/paritytech/substrate/pull/7780/commits/6e3c3366fbb9137c6391cffa5b8517a4a0ace75b

If anyone finds that controversial feel free to revert